nixos/modules/services/hardware/nvidia-container-toolkit.nix

```diff
@@ -0,0 +1,84 @@
+{ config, lib, pkgs, ... }:
+
+{
+
+  options = {
+
+    hardware.nvidia-container-toolkit.enable = lib.mkOption {
+      default = false;
+      type = lib.types.bool;
+      description = lib.mdDoc ''
+        Enable the NVIDIA Container Toolkit (nvidia-ctk) service that
+        generates a Container Device Interface (CDI) configuration file
+        for auto-detected NVIDIA devices, so that CDI-compliant container
+        runtimes can use the forwarded host GPU's.
+      '';
+    };
+
+  };
+
+  config = let
+    generatedCdiConfigPlacement = "/var/run/cdi/nvidia-container-toolkit.json";
+  in {
+    systemd.services.nvidia-container-toolkit = lib.mkIf config.hardware.nvidia-container-toolkit.enable {
+      description = "Container Device Interface (CDI) for Nvidia generator";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "systemd-udev-settle.service" ];
+      serviceConfig = {
+        ExecStart = let
+          script = pkgs.writeScriptBin "nvidia-cdi-generator" ''
+            #! ${pkgs.runtimeShell}
+            mkdir -p /var/run/cdi
```

Comment:
Is `/var/run` hard-coded somewhere upstream? We generally do not want to manage this kind of directories using scripts. Instead we rely on (`man 5 systemd.exec`) `{Runtime,State,...}Directory` (corresponding to `/run`, `/var/lib`, etc and different persistence semantics). In the longer term it'll probably also be easier to migrate to a systemd-agnostic/abstract service layer if we actually had modeled all of these directories declaratively

Reply:
I was thinking in using `/run`, but the spec mentions `/var/run` specifically. Although it's a symlink I'd prefer if we can be 1:1 with the spec: https://github.com/cncf-tags/container-device-interface/blob/7b540ad63013eafbb6779fd06e3235653528d4ae/pkg/cdi/spec-dirs.go#L26-L31.

It's also configurable on some runtimes (https://github.com/cncf-tags/container-device-interface/tree/7b540ad63013eafbb6779fd06e3235653528d4ae#how-to-configure-cdi), but I think it would be good to keep us as close as possible to the spec.